bouncer.go

```diff
@@ -266,6 +315,157 @@ type Login struct {
 	Expire string `json:"expire"`
 }
 
+type SiteVerifyResponse struct {
+	Success     bool      `json:"success"`
+	Action      string    `json:"action"`
+	ChallengeTS time.Time `json:"challenge_ts"`
+	Hostname    string    `json:"hostname"`
+	ErrorCodes  []string  `json:"error-codes"`
+}
+
+type ValidationResponse struct {
+	Success bool   `json:"success"`
+	Message string `json:"message"`
+}
+
+func handleBanResponseSoft(bouncer *Bouncer, rw http.ResponseWriter, req *http.Request) {
+	if bouncer.forbidOnFailure {
+		rw.WriteHeader(http.StatusForbidden)
+		return
+	}
+	bouncer.next.ServeHTTP(rw, req)
+}
+
+func handleBanResponseForced(bouncer *Bouncer, rw http.ResponseWriter, req *http.Request) {
```

Comment:
not understand why you create a function for one line

Reply:
this behaviour is required multiple times, which means it got to be a function